dl1_data_handler/writer.py

```diff
@@ -1,929 +1,893 @@
+
+                    output_filename = self._get_next_filename(
+                        output_filename,
+                        output_file_count)
+
+                    # Create a new Data Dumper pointing at a new file
+                    # and write file-level data
+                    # Will flush + finalize + close file owned by
+                    # previous data dumper
+                    data_dumper = self.data_dumper_class(
+                        output_filename, **self.data_dumper_settings)
+
+                    # Write all file-level data if not present
+                    # Or compare to existing data if already in file
+                    example_event = next(event_source._generator())
+                    subarray = example_event.inst.subarray
+                    mcheader = example_event.mcheader
+                    data_dumper.prepare_file(filename, subarray, mcheader)
+
+            if self.save_mc_events:
+                for mc_event in event_source.file_.iter_mc_events():
+                    try:
+                        data_dumper.dump_mc_event(mc_event, event_source.file_.header['run'])
+                    except IOError:
+                        logger.error("Failed to write event from file "
+                                     "{}, skipping...".format(
+                            filename))
+                        break
+
+                    # Check whether to create another file
+                    max_events_reached = ((self.events_per_file is not None) and (
+                            data_dumper.event_index - 1 >= self.events_per_file))
+
+                    max_size_reached = ((self.output_file_size is not None) and (
+                            os.path.getsize(
+                                data_dumper.output_filename) > self.output_file_size))
+
+                    if max_events_reached or max_size_reached:
+                        # Reset event count and increment file count
+                        output_file_count += 1
+
+                        output_filename = self._get_next_filename(
+                            output_filename,
+                            output_file_count)
+
+                        # Create a new Data Dumper pointing at a new file
+                        # and write file-level data
+                        data_dumper = self.data_dumper_class(
+                            output_filename, **self.data_dumper_settings)
+
+                        # Write all file-level data if not present
+                        # Or compare to existing data if already in file
+                        example_event = next(event_source._generator())
+                        subarray = example_event.inst.subarray
+                        temp = io.DataContainer()
+                        event_source.fill_mc_information(temp, mc_event)
+                        mcheader = temp.mcheader
+                        data_dumper.prepare_file(filename, subarray, mcheader)
```

Comment:
What happen here? Why all the lines where deleted and then added? I can't see the changes.

Reply:
I think there were 3 merge conflicts here that I corrected using GitHub's tool. I am not sure why the entire code was re-added.

Reply:
Can you point me to your changes please?